File: Gelman_BDA_ARM/arm/7.4.1.R
library ("foreign")
library ("arm")
wells <- read.table ("../doc/gelman/ARM_Data/arsenic/wells.dat", header=TRUE)
attach(wells)

fit.1 <- glm (switch ~ dist, family=binomial(link="logit"))
print (summary(fit.1))

sim.1 <- sim (fit.1, n.sims=1000)
plot (sim.1$coef[,1], sim.1$coef[,2], xlab=expression(beta[0]),
  ylab=expression(beta[1]), pch=20)

  

File: Gelman_BDA_ARM/arm/7.4.1.py
import numpy 
from matplotlib import pyplot as plt
from matplotlib import rc
import scikits.statsmodels.api as sm
from scipy import stats
import sim

data = numpy.loadtxt("../doc/gelman/ARM_Data/arsenic/wells.dat",  
                     usecols = (1,2,3,4,5), 
                     skiprows=1)

exog = data[:,2]
endog = data[:,0]

exog = sm.add_constant(exog, prepend=True)

logit_mod = sm.Logit(endog, exog)
logit_res = logit_mod.fit()

[beta, sigma] = sim.sim_glm(logit_res, 1000)

print numpy.mean(beta[:,0])
print numpy.mean(beta[:,1])

plt.plot(beta[:,0], beta[:,1], '.')
plt.xlabel('beta_0')
plt.ylabel('beta_1')
plt.show()
